lib/src/migrators/module/references.dart

```diff
@@ -340,21 +343,31 @@ class _ReferenceVisitor extends RecursiveAstVisitor {
         _importer = result.item1;
         var oldLibraryUrl = _libraryUrl;
         var url = result.item2.span.sourceUrl;
-        if (_importer != oldImporter) _libraryUrl ??= url;
-
+        if (_importer != oldImporter &&
+            !url.toString().endsWith('.import.scss') &&
+            !url.toString().endsWith('.import.sass')) {
+          _libraryUrl ??= url;
+        }
+        var oldRuleUrl = _currentRuleUrl;
+        _currentRuleUrl = import.url;
         visitStylesheet(result.item2);
         var importSource = ImportSource(url, import);
         for (var declaration in _declarationSources.keys.toList()) {
           var source = _declarationSources[declaration];
-          if (source.url == url &&
-              (source is CurrentSource || source is ForwardSource)) {
-            _declarationSources[declaration] = importSource;
+          if (source.url == url) {
+            if (source is CurrentSource || source is ForwardSource) {
+              _declarationSources[declaration] = importSource;
+            } else if (source is ImportOnlySource) {
+              _declarationSources[declaration] =
+                  ImportSource.fromImportOnlyForward(source);
+            }
```

Comment:
IME it helps to make complex logic like this less nested for reading and debugging.

Some suggestions would be:
* on line 333, instead of having all of the logic wrapped in `if (import is DynamicImport) {}`, we could `continue` on the false condition `if (import is! DynamicImport) continue;` and eliminate a whole level of nesting.
* same on line 357, `if (source.url != url) continue;`

Reply:
Done. For the `DynamicImport` check, continuing if it's not one doesn't work because Dart doesn't recognize that `import` should be a `DynamicImport` for the rest of the loop, so instead I added a `whereType` condition to the for loop.